### PR TITLE
Fix request exceptions using URL instead of string (closes #16)

### LINKED ISSUE #16: TypeError: Cannot read property 'startsWith' of undefined (url.hostname)
Problem
--------

When I use `URL` instead of `string` as the url parameter for `http(s).request()`, this error will occur.

Cause
------
In `bindHttpMethod.js`, it checks only the first parameter is `string`, we should check if it is `URL` as well.

Stack trace
-----------
```text
internal/url.js:1259
    hostname: url.hostname.startsWith('[') ?
                           ^

TypeError: Cannot read property 'startsWith' of undefined
    at urlToOptions (internal/url.js:1259:28)
    at request (https.js:279:15)
    at Object.args [as request] (./node_modules/global-agent/dist/utilities/bindHttpMethod.js:55:14)
```

### PR DESCRIPTION
**Pull request by houtianze**

### CODE REVIEW

**Comment by gajus** [comment]
Can we have a test case too, please?

File: src/utilities/bindHttpMethod.js [hunk 1]
```diff
@@ -13,7 +13,8 @@ export default (originalMethod: Function, agent: AgentType, forceGlobalAgent: bo
     let options;
     let callback;
 
-    if (typeof args[0] === 'string') {
+    // the url parameter can be string or URL
```
Comment (gajus) [inline src/utilities/bindHttpMethod.js]: The comment is superfluous. That is literally what is expressed in the next line.

Comment (houtianze) [reply src/utilities/bindHttpMethod.js]: It might be, but what it serves as a reminder, so that in the future this kind of careless bug should be avoided.

**Comment by houtianze** [comment]
> Can we have a test case too, please?

No.
- There is no existing test case for this file in the first place.
- It's not really logic change, but API conformance, and it's way obvious. I'd expect the code to have that level of extensive test coverage already if a test case is to be written for this line of change.

**Pull request merged**